LoopKit/DeviceAlert.swift

```diff
@@ -79,15 +79,38 @@ public struct DeviceAlert {
     /// An alert's "identifier" is a tuple of `managerIdentifier` and `alertIdentifier`.  It's purpose is to uniquely identify an alert so we can
     /// find which device issued it, and send acknowledgment of that alert to the proper device manager.
     public let identifier: Identifier
-        
-    public init(identifier: Identifier, foregroundContent: Content?, backgroundContent: Content?, trigger: Trigger) {
+
+    // Name of the sound file suitable for using to play back
+    public struct SoundName: RawRepresentable, Equatable, CustomStringConvertible {
+        public init(rawValue: String) { self.rawValue = rawValue}
+        public var description: String { return rawValue }
+        public let rawValue: String
+    }
+    public let soundName: SoundName?
+    
+    public init(identifier: Identifier, foregroundContent: Content?, backgroundContent: Content?, trigger: Trigger, soundName: SoundName? = nil) {
         self.identifier = identifier
         self.foregroundContent = foregroundContent
         self.backgroundContent = backgroundContent
         self.trigger = trigger
+        self.soundName = soundName
     }
 }
 
+// Special name to mean "vibrate"
+public extension DeviceAlert.SoundName {
```

Comment:
Seems like making `SoundName` an enum might be more natural here?
```
enum AlertSoundAction {
    case vibrate
    case silence
    case sound(name: String)
}
```

Reply:
Heh, I had it this way at first (before I put up the PR), but it felt like overkill to me, so I made it just a `typealias`.  Then @mpangburn had suggested making a first-class type (struct).  I guess you all would like it the first way I had it :).